src/convolution.jl

```diff
@@ -0,0 +1,95 @@
+"""
+    convolve(d1::T, d2::T) where T<:Distribution -> Distribution
```

Comment:
There might be some closed-forms for different distributions:

```suggestion
    convolve(d1::D1, d2::D2) where {D1<:Distribution, D2 <: Distribution} -> Distribution
```

Reply:
At the moment though they are only defined for the same distribution. Might it be best to leave the generalised form for when the methods with two different distributions are added?

Reply:
Ok fine by me, this is not a blocking point for the PR